src/Enzo/enzo_EnzoMethodInference.cpp

```diff
@@ -0,0 +1,1488 @@
+
+void EnzoLevelArray::interpolate_
+(enzo_float * a_f,
+ int mfx, int mfy, int mfz,
+ int nfx, int nfy, int nfz,
+ int efx, int efy, int efz,
+ const enzo_float * a_c,
+ int mcx, int mcy, int mcz,
+ int ncx, int ncy, int ncz,
+ int ecx, int ecy, int ecz)
+{
+  // initialize index offsets, d[0]=0, d[1] = dx, d[2]=dy, d[3]=dx+dy,
+  // etc.
+  const int dx = 1;
+  const int dy = mcx;
+  const int dz = mcx*mcy;
+  int d[8];
+  for (int iz=0; iz<2; iz++) {
+    for (int iy=0; iy<2; iy++) {
+      for (int ix=0; ix<2; ix++) {
+        const int i=ix+2*(iy+2*iz);
+        d[i] = ix*dx + iy*dy + iz*dz;
+      }
+    }
+  }
+
+  const enzo_float w[2] = { 0.75, 0.25 };
+
+  if (cello::rank() == 1) {
+    for (auto ifx=0; ifx<nfx; ifx++) {
+      auto iex = ifx+(1-efx);
+      auto icx = iex/2;
+      auto wx0 = w[iex&1];
+      auto wx1 = 1.0-wx0;
+      auto i_f = ifx;
+      auto i_c = icx;
+      a_f[i_f] = (wx0*a_c[i_c+d[0]] +
+                  wx1*a_c[i_c+d[1]]);
+    }
+  } else if (cello::rank() == 2) {
+    for (auto ify=0; ify<nfy; ify++) {
+      auto iey = ify+(1-efy);
+      auto icy = iey/2;
+      auto wy0 = w[iey&1];
+      auto wy1 = 1.0-wy0;
+      for (auto ifx=0; ifx<nfx; ifx++) {
+        auto iex = ifx+(1-efx);
+        auto icx = iex/2;
+        auto wx0 = w[iex&1];
+        auto wx1 = 1.0-wx0;
+        auto i_f = ifx + mfx*ify;
+        auto i_c = icx + mcx*icy;
+        a_f[i_f] = (wy0*wx0*a_c[i_c+d[0]] +
+                    wy0*wx1*a_c[i_c+d[1]] +
+                    wy1*wx0*a_c[i_c+d[2]] +
+                    wy1*wx1*a_c[i_c+d[3]]);
+      }
+    }
+  } else if (cello::rank() == 3) {
+    for (auto ifz=0; ifz<nfz; ifz++) {
+      auto iez = ifz+(1-efz);
+      auto icz = iez/2;
+      auto wz0 = w[iez&1];
+      auto wz1 = 1.0-wz0;
+      for (auto ify=0; ify<nfy; ify++) {
+        auto iey = ify+(1-efy);
+        auto icy = iey/2;
+        auto wy0 = w[iey&1];
+        auto wy1 = 1.0-wy0;
+        for (auto ifx=0; ifx<nfx; ifx++) {
+          auto iex = ifx+(1-efx);
+          auto icx = iex/2;
+          auto wx0 = w[iex&1];
+          auto wx1 = 1.0-wx0;
+          auto i_f = ifx + mfx*(ify + mfy*ifz);
+          auto i_c = icx + mcx*(icy + mcy*icz);
+          a_f[i_f] = (wz0*wy0*wx0*a_c[i_c+d[0]] +
+                      wz0*wy0*wx1*a_c[i_c+d[1]] +
+                      wz0*wy1*wx0*a_c[i_c+d[2]] +
+                      wz0*wy1*wx1*a_c[i_c+d[3]] +
+                      wz1*wy0*wx0*a_c[i_c+d[4]] +
+                      wz1*wy0*wx1*a_c[i_c+d[5]] +
+                      wz1*wy1*wx0*a_c[i_c+d[6]] +
+                      wz1*wy1*wx1*a_c[i_c+d[7]]);
+        }
+      }
+    }
+  }
+}
+
+//----------------------------------------------------------------------
+
+void EnzoLevelArray::copy_
+(enzo_float * a_dst,
+ int mdx, int mdy, int mdz, int ndx,int ndy, int ndz, int edx, int edy, int edz,
+ const enzo_float * a_src,
+ int msx, int msy, int msz, int nsx,int nsy, int nsz, int esx, int esy, int esz)
```

Comment:
It looks like some of these parameters aren't used. For example `edx, edy, edz, esx, esy, esz, nsx, nsy, nsz`